### PR TITLE
Fix build issue with KafkaDataset

### LINKED ISSUE #17210: NotFoundError: Op type not registered 'KafkaDataset' in binary.
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: - 
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: ('unknown', '1.6.0-rc1')
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: 0.8.0
- **GCC/Compiler version (if compiling from source)**: 5.4.0
- **CUDA/cuDNN version**: -
- **GPU model and memory**: -
- **Exact command to reproduce**:

~~~python
from tensorflow.contrib.kafka.python.ops import kafka_dataset_ops
from tensorflow.python.data.ops import iterator_ops
from tensorflow.python.framework import dtypes
from tensorflow.python.ops import array_ops

topics = array_ops.placeholder(dtypes.string, shape=[None])
num_epochs = array_ops.placeholder(dtypes.int64, shape=[])
batch_size = array_ops.placeholder(dtypes.int64, shape=[])

repeat_dataset = kafka_dataset_ops.KafkaDataset(
    topics, group="test", eof=True).repeat(num_epochs)
batch_dataset = repeat_dataset.batch(batch_size)

iterator = iterator_ops.Iterator.from_structure(batch_dataset.output_types)
init_op = iterator.make_initializer(repeat_dataset)
init_batch_op = iterator.make_initializer(batch_dataset)
get_next = iterator.get_next()
~~~

ENV: https://pastebin.com/89aihba7

### Describe the problem
Getting `NotFoundError` even with `TF_NEED_KAFKA=1` and `--define with_kafka_support=true`:
~~~
NotFoundError: Op type not registered 'KafkaDataset' in binary running on 68a9f992375e. Make sure the Op and Kernel are registered in the binary running in this process.
~~~
`KafkaDataset` was merged into master last month. Is there something missing that needs to be done in order to utilize the new op? 

### Source code / logs
https://pastebin.com/5Vy6b6kf

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address the issue raised in #17210 where error of `NotFoundError: Op type not registered 'KafkaDataset' in binary.` returned from kafka ops.

The issue was that the inclusion of kafka ops was removed due to the conflict merge from the other PR. This fix fixes the issue.

This fix fixes #17210.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/core/kernels/BUILD [hunk 1]
```diff
@@ -1590,6 +1590,17 @@ cc_header_only_library(
     deps = [":lookup"],
 )
 
+cc_library(
```
Comment (mrry) [inline tensorflow/core/kernels/BUILD]: This new library shouldn't be necessary. The `dataset.h` header is now in `tensorflow/core/framework` and can be included via the `//tensorflow/core:framework_headers_lib` target.

You might want to check out the targets in `//tensorflow/contrib/data/...` that build dataset implementations as custom op libraries. This has changed in the last month or so, as we've refactored things to make separate compilation possible for dataset implementations.

**Review by mrry — APPROVED** [review]
Neat! Thanks for making those changes. LGTM.

**Comment by yongtang** [comment]
Thanks @mrry for the help 👍 . Some of the tests are failing. Seems to be some difference between clang and gcc (it runs fine on my local machine with gcc + ubuntu 16.04). I am looking into the issue now.

**Comment by yongtang** [comment]
The reason of the previous test failure was that, `includes` in BUILD file adds include path with `-isystem` (vs. `-I`). That makes clang unhappy. The issue has been fixed by replacing `includes` with `copt`.

All tests passed now 🎉

**Pull request merged**